extmod/uasyncio/core.py

```diff
@@ -145,26 +146,32 @@ def create_task(coro):
     _task_queue.push(t)
     return t
 
-
 # Keep scheduling tasks until there are none left to schedule
 def run_until_complete(main_task=None):
     global cur_task
     excs_all = (CancelledError, Exception)  # To prevent heap allocation in loop
     excs_stop = (CancelledError, StopIteration)  # To prevent heap allocation in loop
+    _io_queue.wait_io_event(0)
```

Comment:
is this still needed, now that `wait_io_event()` is always called below?

Reply:
You're right. I removed it. I thought I needed it for one of the tests, but they all pass without it.